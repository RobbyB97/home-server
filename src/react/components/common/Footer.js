import React from 'react'


export class Footer extends React.Component {
    
    render() {
        return (
            <div id="footer">
                <div class="footer__logos">

                </div>
                <div class="footer__copyright">
                    
                </div>
            </div>
        )
    }
}


export default Footer