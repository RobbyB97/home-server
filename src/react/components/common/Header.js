import React from 'react'
import {NavLink} from 'react-router-dom'
import {connect} from 'react-redux'

import {toggleNav, toggleNavOff} from '../../actions/navigation'
import {setTitle} from '../../actions/title'
import {homeOn} from '../../actions/home'


const Header = ({nav, dispatch}) => (
	<header id="header">
		<div className="header__logo">
	  		<NavLink to="/react/" onClick={() => {
				dispatch(setTitle(''))
				dispatch(toggleNavOff())
				dispatch(homeOn())
	  		}}>
			<img src='/assets/images/favicon.png' alt='Home'/> 
	  		</NavLink>
		</div>

		<button className="header__nav-button" onClick={() => {
	  		dispatch(toggleNav())
		}}>
	  		{!nav ? '☰':'✕'}
		</button>    
  	</header>
) 

const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav
  	}
}

export default connect(mapStateToProps)(Header)