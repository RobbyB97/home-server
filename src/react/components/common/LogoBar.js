import React from 'react'

const LogoBar = () => (
    <div id="logo-bar">
        <ul>
            <li>
                <a href="https://www.github.com/robbyb97" target="_blank">
                    <img src="/dist/images/icons/github.svg" />
                </a>
            </li>
            
            <li>
                <a href="https://www.linkedin.com/in/robbyb97" target="_blank">
                    <img src="/dist/images/icons/linkedin.png" />
                </a>
            </li>
        </ul>
    </div>
)

export default LogoBar