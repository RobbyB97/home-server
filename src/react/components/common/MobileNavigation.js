import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {toggleNav, toggleNavOff} from '../../actions/navigation'
import {setTitle} from '../../actions/title'
import {homeOn, homeOff} from '../../actions/home'


const MobileNavigation = ({nav, dispatch, home}) => (
    <div id="mobile-navigation" className={nav ? 'active-nav':''}>
        <div className="mobile-navigation__background">
            <ul className="mobile-navigation__intra-nav">
                <li>
                    <NavLink to="/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle(''))
                        dispatch(homeOn())
                    }}>Home</NavLink>
                </li>

                <li>
                    <NavLink to="/skills/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Skills'))
                        dispatch(homeOff())
                    }}>Skills</NavLink>
                </li>

                <li>
                    <NavLink to="/experience/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Experience'))
                        dispatch(homeOff())
                    }}>Experience</NavLink>
                </li>

                <li>
                    <NavLink to="/projects/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Projects'))
                        dispatch(homeOff())
                    }}>Projects</NavLink>
                </li>

                <li>
                    <NavLink to="/contact/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Contact'))
                        dispatch(homeOff())
                    }}>Contact</NavLink>
                </li>
            </ul>
        </div>
    </div>
)

const mapStateToProps = (state) => {
    return {
        nav: state.navigation.nav,
        home: state.home.home
    }
}

export default connect(mapStateToProps)(MobileNavigation)