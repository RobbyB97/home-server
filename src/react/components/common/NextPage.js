import React from 'react'
import {NavLink} from 'react-router-dom'


export class NextPage extends React.Component {
    

    render() {
        return (
            <div className={this.props.float ? "nextPage float" : "nextPage"}>
                <NavLink 
                    to={this.props.route ? this.props.route : '/'}
                >
                    <p className="nextPage__text">
                        {this.props.text && this.props.text} â†’
                    </p>
                </NavLink>
            </div>
        )
    }
}


export default NextPage