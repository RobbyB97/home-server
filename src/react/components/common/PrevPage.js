import React from 'react'
import {NavLink} from 'react-router-dom'


export class PrevPage extends React.Component {
    

    render() {
        return (
            <div className="prevpage">
                {this.props.text &&
                    <p className="prevpage__text">
                        {this.props.text}
                    </p>
                }
                <NavLink 
                    to={this.props.route ? this.props.route : '/'}
                >
                    <div className="prevpage__arrow">
                        <img src="/dist/images/icons/down_arrow_white.png"/>
                    </div>
                </NavLink>
            </div>
        )
    }
}


export default PrevPage