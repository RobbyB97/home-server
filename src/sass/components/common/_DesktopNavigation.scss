#desktop-navigation {
    display: none;

    @media #{$bp-desktop} {
        display: block;
        flex-grow: 1;
        position: fixed;
        z-index: 10;
        right: 2.2rem;
        top: 1.2rem;
        background: transparent;
        font-size: $s-font;
        height: 3rem;

        ul {
            display: flex;
            flex-direction: row;
            margin-top: 0;
            margin-bottom: 0;

            li {
                list-style-type: none;
                margin-right: $m-font;

                a {
                    color: white;
                    transition: 200ms;
                    text-decoration: none;

                    &:hover {
                        text-decoration: none;
                        border-bottom: 2px solid $purple;
                    }
                }

                .active {
                    border-bottom: 1px solid white;
                    text-decoration: none;
                }
            }
        }
    }

    @media #{$bp-large} {
        ul {
            margin-top: 12px ;
            margin-bottom: 12px;
        }
    }
}


.desktop-navigation__link {
    margin-right: 1rem;

    a {
        width: $m-font;
        height: $m-font;
        border-radius: 50%;
        padding-bottom: 2px;
        transition: 200ms;
        
        &:hover {
            border-bottom: 0;
        }                        
    }

    &.linkedin {
        margin-right: 0 !important;

        a > div {
            background: url($linkedin);
            background-size: 100% 100%;
            width: $m-font;
            height: $m-font;

            &:hover {
                background: url($linkedin-purple);
                background-size: 100% 100%;
            }   
        }
    }

    &.github {
                    
        a > div {
            background: url($github);
            background-size: 100% 100%;
            width: $m-font;
            height: $m-font;

            &:hover {
                background: url($github-purple);
                background-size: 100% 100%;
            }
        }
    }
}