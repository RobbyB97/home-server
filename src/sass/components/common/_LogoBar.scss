#logo-bar {
    position: fixed;
    top: 14px;
    right: 4rem;
    z-index: 100;
    display: block;

    img {
        width: 2em;
        max-width: 2em;
        height: auto;
        border-radius: 50%;
        vertical-align: top;
    }
    
    ul {
        display: flex;
        flex-direction: row;
        margin: 0;
        
        li {
            display: inline-block;
            margin-right: 1rem;
            
            &:last-child {
                margin-right: 0;
            }
        }
    }
    
    @media #{$bp-tablet} {
        display: none;
    }
}