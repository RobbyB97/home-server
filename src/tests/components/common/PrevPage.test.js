import React from 'react'
import {shallow} from 'enzyme'

import {PrevPage} from '../../../react/components/common/PrevPage'
import nextpages from '../../fixtures/nextpages'


test('Render PrevPage with no props', () => {
    const wrapper = shallow(<PrevPage />)

    expect(wrapper).toMatchSnapshot()
})


test('Render PrevPage with text', () => {
    const wrapper = shallow(<PrevPage {...nextpages[0]}/>)

    expect(wrapper).toMatchSnapshot()
})


test('Render PrevPage with route', () => {
    const wrapper = shallow(<PrevPage {...nextpages[1]}/>)

    expect(wrapper).toMatchSnapshot()
})


test('Render PrevPage with all props', () => {
    const wrapper = shallow(<PrevPage {...nextpages[2]}/>)

    expect(wrapper).toMatchSnapshot()
})