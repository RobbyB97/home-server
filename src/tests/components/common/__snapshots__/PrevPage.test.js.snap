// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PrevPage with all props 1`] = `
<div
  className="prevPage"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="Route!"
  >
    <p
      className="prevPage__text"
    >
      ← 
      Text!
    </p>
  </NavLink>
</div>
`;

exports[`Render PrevPage with no props 1`] = `
<div
  className="prevPage"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="/"
  >
    <p
      className="prevPage__text"
    >
      ← 
    </p>
  </NavLink>
</div>
`;

exports[`Render PrevPage with route 1`] = `
<div
  className="prevPage"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="Route!"
  >
    <p
      className="prevPage__text"
    >
      ← 
    </p>
  </NavLink>
</div>
`;

exports[`Render PrevPage with text 1`] = `
<div
  className="prevPage"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="/"
  >
    <p
      className="prevPage__text"
    >
      ← 
      Text!
    </p>
  </NavLink>
</div>
`;
